android/ru/ra66it/qrcodegen/QrCode.java

```diff
@@ -0,0 +1,870 @@
+     * and {@link #encodeBinary(byte[], Ecc)}.</p>
+     *
+     * @param segs the segments to encode
+     * @param ecl  the error correction level to use (not {@code null}) (boostable)
+     * @return a QR Code (not {@code null}) representing the segments
+     * @throws NullPointerException if the list of segments, any segment, or the error correction level is {@code null}
+     * @throws DataTooLongException if the segments fail to fit in the
+     *                              largest version QR Code at the ECL, which means they are too long
+     */
+    public static QrCode encodeSegments(List<QrSegment> segs, Ecc ecl) {
+        return encodeSegments(segs, ecl, MIN_VERSION, MAX_VERSION, -1, true);
+    }
+
+
+    /**
+     * Returns a QR Code representing the specified segments with the specified encoding parameters.
+     * The smallest possible QR Code version within the specified range is automatically
+     * chosen for the output. Iff boostEcl is {@code true}, then the ECC level of the
+     * result may be higher than the ecl argument if it can be done without increasing
+     * the version. The mask number is either between 0 to 7 (inclusive) to force that
+     * mask, or &#x2212;1 to automatically choose an appropriate mask (which may be slow).
+     * <p>This function allows the user to create a custom sequence of segments that switches
+     * between modes (such as alphanumeric and byte) to encode text in less space.
+     * This is a mid-level API; the high-level API is {@link #encodeText(CharSequence, Ecc)}
+     * and {@link #encodeBinary(byte[], Ecc)}.</p>
+     *
+     * @param segs       the segments to encode
+     * @param ecl        the error correction level to use (not {@code null}) (boostable)
+     * @param minVersion the minimum allowed version of the QR Code (at least 1)
+     * @param maxVersion the maximum allowed version of the QR Code (at most 40)
+     * @param mask       the mask number to use (between 0 and 7 (inclusive)), or &#x2212;1 for automatic mask
+     * @param boostEcl   increases the ECC level as long as it doesn't increase the version number
+     * @return a QR Code (not {@code null}) representing the segments
+     * @throws NullPointerException     if the list of segments, any segment, or the error correction level is {@code null}
+     * @throws IllegalArgumentException if 1 &#x2264; minVersion &#x2264; maxVersion &#x2264; 40
+     *                                  or &#x2212;1 &#x2264; mask &#x2264; 7 is violated
+     * @throws DataTooLongException     if the segments fail to fit in
+     *                                  the maxVersion QR Code at the ECL, which means they are too long
+     */
+    public static QrCode encodeSegments(List<QrSegment> segs, Ecc ecl, int minVersion, int maxVersion, int mask, boolean boostEcl) {
+        Objects.requireNonNull(segs);
+        Objects.requireNonNull(ecl);
```

Comment:
these lines are doing nothing, u gotta set the return value to a variable
```suggestion
        segs = Objects.requireNonNull(segs);
        ecl = Objects.requireNonNull(ecl);
```